app/controllers/HasUtrController.scala

```diff
@@ -16,57 +16,76 @@
+                                  sessionRepository: SessionRepository,
+                                  navigator: Navigator,
+                                  identify: IdentifierAction,
+                                  getData: DataRetrievalAction,
+                                  requireData: DataRequiredAction,
+                                  formProvider: HasUtrFormProvider,
+                                  val controllerComponents: MessagesControllerComponents,
+                                  corporationTaxView: HasUtrCorporationTaxView,
+                                  selfAssessmentView: HasUtrSelfAssessmentView,
+                                  partnershipView: HasUtrPartnershipView
+                                 )(implicit ec: ExecutionContext)
+  extends FrontendBaseController with I18nSupport with AnswerExtractor {
 
   val form = formProvider()
 
   def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
     implicit request =>
+      getAnswer(BusinessTypePage) {
+        businessType =>
 
-      val preparedForm = request.userAnswers.get(HasUtrPage) match {
-        case None => form
-        case Some(value) => form.fill(value)
-      }
+          val preparedForm = request.userAnswers.get(HasUtrPage) match {
+            case None => form
+            case Some(value) => form.fill(value)
+          }
 
-      Ok(view(preparedForm, mode))
+          Ok(renderView(businessType, preparedForm, mode))
+      }
   }
 
   def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
+      getAnswerAsync(BusinessTypePage) {
+        businessType =>
 
-      form.bindFromRequest().fold(
-        formWithErrors =>
-          Future.successful(BadRequest(view(formWithErrors, mode))),
+          form.bindFromRequest().fold(
+            formWithErrors =>
+              Future.successful(BadRequest(renderView(businessType, formWithErrors, mode))),
 
-        value =>
-          for {
-            updatedAnswers <- Future.fromTry(request.userAnswers.set(HasUtrPage, value))
-            _              <- sessionRepository.set(updatedAnswers)
-          } yield Redirect(navigator.nextPage(HasUtrPage, mode, updatedAnswers))
-      )
+            value =>
+              for {
+                updatedAnswers <- Future.fromTry(request.userAnswers.set(HasUtrPage, value))
+                _ <- sessionRepository.set(updatedAnswers)
+              } yield Redirect(navigator.nextPage(HasUtrPage, mode, updatedAnswers))
+          )
+      }
   }
+
+  private def renderView(businessType: BusinessType, form: Form[_], mode: Mode)(implicit request: Request[_]) =
+    businessType match {
+      case LimitedCompany | AssociationOrTrust => corporationTaxView(form, mode)
+      case Llp | Partnership => partnershipView(form, mode)
+      case SoleTrader | Individual => selfAssessmentView(form, mode)
```

Comment:
Please make `Individual` go to an error.

Reply:
Done in a new commit